llama_stack/core/routing_tables/models.py

```diff
@@ -63,7 +63,12 @@ async def get_model(self, model_id: str) -> Model:
 
     async def get_provider_impl(self, model_id: str) -> Any:
         model = await lookup_model(self, model_id)
-        return self.impls_by_provider_id[model.provider_id]
+        try:
+            return self.impls_by_provider_id[model.provider_id]
+        except KeyError as e:
+            raise ValueError(
```

Comment:
maybe make this message a bit vague since it is only one possibility. I encountered this also when I had not specified OLLAMA_URL which means ollama wasn't running. You could argue my registry was out of date, but it is kind of tricky for the user to know.